backend/tests/test_api/test_missions_api.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+
+import pytest
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+from requests import Response
+
+from flotilla.echo.requests import get_echo_requests
+from flotilla.main import app
+from tests.mocks.echo_requests_mock import EchoRequestsMock
+
+CURRENT_DIR: str = os.path.dirname(os.path.abspath(__file__))
+
+file_missions: str = f"{CURRENT_DIR}/data/missions.json"
+with open(file_missions) as file:
+    missions_json: dict = json.load(file)
+
+
+@pytest.mark.parametrize(
+    "request_status_code, request_json, expected_status_code",
+    [
+        (200, missions_json, 200),
+        (404, dict(), 404),
+    ],
+)
+def test_read_missions(
+    test_app: FastAPI,
+    request_status_code: int,
+    request_json: dict,
+    expected_status_code: int,
+):
+    mock_response = Response()
+    mock_response.status_code = request_status_code
+    mock_response.json = lambda: request_json
+    echo_requests_mock = EchoRequestsMock()
+    echo_requests_mock.add_missions_responses([mock_response])
+
+    def get_echo_requests_mock():
+        yield echo_requests_mock
+
+    app.dependency_overrides[get_echo_requests] = get_echo_requests_mock
```

Comment:
Alternatively drop the helper function altogether and just use the EchoRequestsMock directly
```suggestion
    app.dependency_overrides[get_echo_requests] = EchoRequestsMock()
```

Reply:
I believe that it needs some sort of generator function. This is based on FastAPIs example for how to share a database in the different endpoints. The suggestion don't work since the dependency needs to be a callable function. Also tested adding `__call__` to EchoRequestMock and use the suggestion, but I got a 422 status code indicating that fastAPI is not able to handle the dependency.

Reply:
My bad, I see the difference now